packages/medusa/src/loaders/custom-api.ts

```diff
@@ -0,0 +1,56 @@
+import { Express, NextFunction, Request, Response, Router } from "express"
+import { ConfigModule } from "../types/global"
+
+enum AllowedMethods {
+  GET = "GET",
+  POST = "POST",
+  DELETE = "DELETE",
+}
+
+type Handler = (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => Promise<void>
+
+export type MedusaRoute = {
+  name: string
+  method: AllowedMethods
+  path: string
+  handler: Handler | Handler[]
+  routes: MedusaRoute[]
+  options: {}
+}
+
+export default ({
+  app,
+  configModule,
+}: {
+  app: Express
+  configModule: ConfigModule
+}) => {
+  configModule.routes.map(async (customRoute) => {
+    let { handler, path, method, routes, name } = customRoute
+
+    if (!method || !path || !handler || !name) {
+      console.warn(`Invalid config for custom route: ${name}`)
+    }
+
+    if (routes) {
+      const subRouter = Router({ mergeParams: true })
+
```

Comment:
suggestion(non-blocking): Add handlers for the route here as well to allow the definition of middleware for all sub-routes. Would be ideal for something like authentication middleware: 
```
Routes: [
    {
      path: "/hello-world",
      handler: [middlewares.authenticate()],
      routes: [
        { path: "/1", handler: [custom1], method: "GET" },
        { path: "/2", handler: [custom2], method: "GET" },
        ...
      ],
    },
  ],
```

```suggestion
if (handler) {
      subRouter.use(handler)
}
```